cpp/include/cudf/io/detail/data_casting.cuh

```diff
@@ -307,67 +342,19 @@ std::unique_ptr<column> parse_data(str_tuple_it str_tuples,
-                         }
-
-                         auto const in_begin = str_tuples[row].first;
-                         auto const in_end   = in_begin + str_tuples[row].second;
-                         auto out_it         = cub::DiscardOutputIterator<>{};
-                         auto const str_process_info =
-                           process_string(in_begin, in_end, out_it, options);
-
-                         // The total number of characters that we're supposed to copy out
-                         auto const num_chars_copied_out =
-                           thrust::distance(out_it, str_process_info.output_processed_end);
-
-                         // If, during parsing, an error occurred or we parsed the null literal ->
-                         // set to null
-                         if (str_process_info.result != data_casting_result::PARSING_SUCCESS) {
-                           sizes[row] = 0;
-                           clear_bit(null_mask, row);
-                         } else {
-                           sizes[row] = num_chars_copied_out;
-                         }
-                       });
-
-    // Compute offsets for the post-processed strings
-    thrust::exclusive_scan(
-      rmm::exec_policy(stream), offsets.begin(), offsets.end(), offsets.begin());
-
-    // Write out post-processed strings (stripping off quotes, replacing escape sequences)
-    rmm::device_uvector<char> chars(offsets.back_element(stream), stream);
-    thrust::for_each_n(rmm::exec_policy(stream),
-                       thrust::make_counting_iterator<size_type>(0),
-                       col_size,
-                       [str_tuples,
-                        chars     = device_span<char>{chars},
-                        offsets   = device_span<size_type>{offsets},
-                        null_mask = static_cast<bitmask_type*>(null_mask.data()),
-                        options] __device__(size_type row) {
-                         if (not bit_is_set(null_mask, row)) { return; }
-
-                         auto const in_begin = str_tuples[row].first;
-                         auto const in_end   = in_begin + str_tuples[row].second;
-                         auto out_it         = &chars[offsets[row]];
-                         process_string(in_begin, in_end, out_it, options);
-                       });
-
-    return make_strings_column(
-      col_size, std::move(offsets), std::move(chars), std::move(null_mask));
+    // this utility calls the functor to build the offsets and chars columns
+    auto [offsets, chars] = cudf::strings::detail::make_strings_children(
+      string_parse<decltype(str_tuples)>{
+        str_tuples, static_cast<bitmask_type*>(null_mask.data()), options},
+      col_size,
+      stream,
+      mr);
+
+    return make_strings_column(col_size,
+                               std::move(offsets),
+                               std::move(chars),
+                               cudf::UNKNOWN_NULL_COUNT,
```

Comment:
Would it be possible to compute the null count in the functor?
I'd like us to try not using `UNKNOWN_NULL_COUNT` in new code.

Reply:
udpated.